source/qdk_package/src/qdk/__init__.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""qdk bundling meta-package.
+
+Design goals:
+    * Provide a single import root `qdk` that exposes bundled quantum tooling as
+        submodules (`qdk.qsharp`, `qdk.widgets`, etc.).
+
+Optional extras:
+    azure   -> installs `azure-quantum`, available as `qdk.azure`.
+    qiskit  -> installs `qiskit`, available as `qdk.qiskit`.
+    jupyter -> installs `qsharp-widgets` + `qsharp-jupyterlab`; exposes `qdk.widgets`.
+
+"""
+
+from __future__ import annotations
+
+from importlib import import_module
+
+# Eagerly check for qsharp presence to provide a clear error message.
+try:
+    import_module("qsharp")
+except ModuleNotFoundError as _ex:
+    raise ImportError(
+        "qdk requires the 'qsharp' package. Install with 'pip install qsharp'."
+    ) from _ex
+
+# Eagerly import qdk.qsharp to ensure qdk.qsharp is always available if qdk is.
+try:
+    from . import qsharp as qsharp
+except Exception as _ex:
+    raise
+
+# Optional: use telemetry hook if present (skipped in stub/mock envs)
+try:
+    import qsharp.telemetry_events.on_qdk_import
+
+    qsharp.telemetry_events.on_qdk_import()
+except Exception:
+    pass
+
+# Some common utilities are lifted to the qdk root.
+from qsharp import code
+from qsharp import (
+    set_quantum_seed,
+    set_classical_seed,
+    dump_machine,
+    dump_circuit,
+    Result,
+    TargetProfile,
+    StateDump,
+    ShotResult,
+    PauliNoise,
+    DepolarizingNoise,
+    BitFlipNoise,
+    PhaseFlipNoise,
+)
+
+__all__ = [
+    "qsharp",
+    "estimator",
+    "openqasm",
+    "widgets_available",
+    "azure_available",
+    "qiskit_available",
+    "require",
+    # utilities
+    "code",
+    "set_quantum_seed",
+    "set_classical_seed",
+    "dump_machine",
+    "dump_circuit",
+    "Result",
+    "TargetProfile",
+    "StateDump",
+    "ShotResult",
+    "PauliNoise",
+    "DepolarizingNoise",
+    "BitFlipNoise",
+    "PhaseFlipNoise",
+]
+
+
+def widgets_available() -> bool:
```

Comment:
I still don't understand what the value is for everything from this line down. Why do we need/want this?

Reply:
removed, see comment below